lib/identity/client.js

```diff
@@ -886,16 +926,99 @@ class Client extends PartialClient {
       )
     }
     // if the admin sharing group will be empty, revoke the share & delete the group
-    const groupName = `tozny.secret.${this.config.realmName}.${
-      this.storage.config.clientId
-    }.${secret.meta.plain.secretType.toLowerCase()}`
+    const groupName = `tozny.secret.${this.storage.config.clientId}.${this.config.realmName}`
     await this.revokeShareBeforeDeleteSecret(groupName, secret.meta.type)
     // delete the secret
     return await this.storage.deleteRecord(
       secret.meta.recordId,
       secret.meta.version
     )
   }
+  /**
+   * isValidToznySecretNamespace
+   * @param {Object} groupName
+   * @return {Boolean} returns true if the secret was successfully removed
+   */
+  async isValidToznySecretNamespace(groupName) {
+    var groupNameSplit = groupName.split('.')
+    if (
+      groupNameSplit.length < 2 ||
+      groupNameSplit[0] != 'tozny' ||
+      groupNameSplit[1] != 'secret'
+    ) {
+      return false
+    }
+    return true
+  }
+  /**
+   * getNamespace
+   * @param {Object} namespace
+   * @return {Object} returns the group based on namespace
+   */
+  async getNamespace(namespace) {
```

Comment:
Because the work is so similar, I think this and the `getOrCreateNamesapce` methods can be combined. Also, this is public on the client, but it is not a function we want users to know or think about really. Namespace is an abstract concept. So for that reason I may pull this out as a protected function defined above the class. Keep the consumer interface a little cleaner with less lower level footguns available.